src/main/kotlin/no/nav/helse/spenn/dao/OppdragStateJooqRepository.kt

```diff
@@ -105,14 +104,25 @@ class OppdragStateJooqRepository(val jooq: DSLContext): OppdragStateRepository {
 
     }
 
-
+    override fun findAllNotAvstemtWithAvstemmingsnokkelNotAfter(avstemmingsnokkelMax: LocalDateTime): List<OppdragState> {
+        return selectOppdragStateJoinAvstemmingOnCondition()
+                .where(AVSTEMMING.AVSTEMT.equal(false))
+                .and(AVSTEMMING.NOKKEL.isNotNull)
+                .and(AVSTEMMING.NOKKEL.le(avstemmingsnokkelMax.toTimeStamp()))
+                .map { it.into(OPPDRAGSTATE).toOppdragState(it.into(AVSTEMMING))}
+    }
 
     private fun selectOppdragStateLeftJoinAvstemmingOnCondition(): SelectOnConditionStep<Record> {
         return jooq.select().from(OPPDRAGSTATE)
                 .leftJoin(AVSTEMMING)
                 .on(OPPDRAGSTATE.ID.equal(AVSTEMMING.OPPDRAGSTATE_ID))
     }
 
+    private fun selectOppdragStateJoinAvstemmingOnCondition(): SelectOnConditionStep<Record> {
```

Comment:
jeg ville bare lagt denne i samme funksjon, for å lettere å kunne se, den er vel ikke brukt noen andre steder?

Reply:
ikke ennå, så, ja, enig